core/sql/common/NAType.cpp

```diff
@@ -670,6 +671,84 @@ short NAType::convertTypeToText(char * text,	   // OUTPUT
 				 displayCaseSpecific);
 }
 
+short NAType::getMyTypeAsHiveText(NAString * outputStr)  // output
+{
+  Lng32		      fs_datatype		= getFSDatatype();
+
+  switch (fs_datatype)
+    {
+    case REC_MIN_F_CHAR_H ... REC_MAX_F_CHAR_H:
+      *outputStr = "string";
+      break;
+
+    case REC_MIN_V_CHAR_H ... REC_MAX_V_CHAR_H:
+      {
+        SQLVarChar * ct = (SQLVarChar*)this;
+        if (ct->wasHiveString())
+          *outputStr = "string";
```

Comment:
Ah, now I see the point of the "wasHiveString" business. It's to recover the original Hive data type on a describe.